src/TrackGenerator.cpp

```diff
@@ -38,13 +38,12 @@ TrackGenerator::~TrackGenerator() {
     delete [] _num_tracks;
     delete [] _num_x;
     delete [] _num_y;
-    delete [] _num_tracks_by_parallel_group;
 
     for (int i = 0; i < _num_azim_2; i++)
       delete [] _tracks[i];
 
     delete [] _tracks;
-    delete [] _tracks_by_parallel_group;
+    delete [] _tracks_array;
```

Comment:
Could we just have one tracks array? It seems a little unnecessary to have two. Perhaps we could then have a macro that allows the tracks to be indexed using the azimuthal and xy index.